avaframe/com1DFA/com1DFA.py

```diff
@@ -84,66 +88,75 @@ def com1DFAMain(avalancheDir, cfgMain, cfgFile='', relThField='', variationDict=
-            log.warning('simulation configuration has been changed since start')
-            cfgUtils.writeCfgFile(avalancheDir, com1DFA, cfg, fileName='%s_butModified' % simHash)
-
-    # Set directory for report
-    reportDir = pathlib.Path(avalancheDir, 'Outputs', modName, 'reports')
-    # Generate plots for all peakFiles
-    plotDict = oP.plotAllPeakFields(avalancheDir, cfgMain['FLAGS'], modName, demData=dem)
-    # write report
-    gR.writeReport(reportDir, reportDictList, cfgMain['FLAGS'], plotDict)
-
-    # read all simulation configuration files and return dataFrame and write to csv
-    standardCfg = cfgUtils.getDefaultModuleConfig(com1DFA, toPrint=False)
-    simDF = cfgUtils.createConfigurationInfo(avalancheDir, standardCfg=standardCfg, writeCSV=True)
-
-    return particlesList, fieldsList, tSave, dem, plotDict, reportDictList
+    # first get already existing simulations
+    simDFOld, simNameOld = cfgUtils.readAllConfigurationInfo(avalancheDir, specDir='')
+    # prepare simulation to run (only the new ones)
+    simDict = prepareVarSimDict(modCfg, inputSimFiles, variationDict, simNameOld=simNameOld)
+
+    # is there any simulation to run?
+    if bool(simDict):
+        reportDictList = []
+        simDF = ''
+        # loop over all simulations
+        for cuSim in simDict:
+
+            # load configuration dictionary for cuSim
+            cfg = simDict[cuSim]['cfgSim']
+
+            # save configuration settings for each simulation
+            simHash = simDict[cuSim]['simHash']
+            cfgUtils.writeCfgFile(avalancheDir, com1DFA, cfg, fileName=cuSim)
+            # append configuration to dataframe
+            simDF = cfgUtils.appendCgf2DF(simHash, cuSim, cfg, simDF)
+
+            # log simulation name
+            log.info('Run simulation: %s' % cuSim)
+
+            # set release area scenario
+            inputSimFiles['releaseScenario'] = simDict[cuSim]['relFile']
+
+            # +++++++++++++++++++++++++++++++++
+            # ------------------------
+            particlesList, fieldsList, tSave, dem, reportDict, cfgFinal, Tcpu = com1DFA.com1DFACore(cfg, avalancheDir,
+                    cuSim, inputSimFiles, outDir, relThField)
+
+            # +++++++++EXPORT RESULTS AND PLOTS++++++++++++++++++++++++
+
+            reportDictList.append(reportDict)
+
+            # export for visulation
+            if cfg['VISUALISATION'].getboolean('writePartToCSV'):
+                outDir = pathlib.Path(avalancheDir, 'Outputs', modName)
+                com1DFA.savePartToCsv(cfg['VISUALISATION']['particleProperties'], particlesList, outDir)
+
+            # create hash to check if config didnt change
+            simHashFinal = cfgUtils.cfgHash(cfgFinal)
+            if simHashFinal != simHash:
+                log.warning('simulation configuration has been changed since start')
+                cfgUtils.writeCfgFile(avalancheDir, com1DFA, cfg, fileName='%s_butModified' % simHash)
```

Comment:
These two lines are not covered by a pytest. However, since we even throw a warning, I think we need to test whether it actually works... -> pytest